Loop/Managers/DeviceDataManager.swift

```diff
@@ -697,8 +789,9 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
             if let sentrySupported = pumpState?.pumpModel?.larger where !sentrySupported {
                 rileyLinkManager.idleListeningEnabled = false
             }
-
             NSUserDefaults.standardUserDefaults().pumpModelNumber = pumpState?.pumpModel?.rawValue
+            self.enableRileyLinkHeartbeatIfNeeded()
```

Comment:
Idle listening will fail to restart ~once every 24 hours since it doesn't restart independently of the app waking.

This was an observation I made early on, and it was relatively recently re-confirmed when someone tried looping using just G4 download before I surfaced the timer tick.